files/nrpe/check_object.py

```diff
@@ -144,6 +169,81 @@ def execute(self):
     except:
       raise
 
+class SwiftPublicAvailability():
+  '''
+  Check Swift API call length by listing contents of a public bucket
+  '''
+  options = dict()
+
+  def __init__(self, options):
+
+    self.creds = OSCredentials(options)
+
+    try:
+      if options.swift_publicurl is not None:
+        self.swift_publicurl = options.swift_publicurl
+      else:
+        self.swift_publicurl = os.environ['swift_publicurl']
+    except KeyError: 
+      raise CredentialsOrParametersMissingException(key='swift_publicurl')
+
+    try:
+      if options.swift_bucket is not None:
+        self.swift_bucket = options.swift_bucket
+      else:
+        self.swift_bucket = os.environ['swift_bucket']
+    except KeyError: 
+      raise CredentialsOrParametersMissingException(key='swift_bucket')
+
+    try:
+      if options.tenant is not None:
+        self.tenant = options.tenant
+      else:
+        self.tenant = os.environ['tenant']
+    except KeyError: 
+      raise CredentialsOrParametersMissingException(key='tenant')
+
+  def list_public_swift_objects(self):
+    """ read a public swift URL with urllib
+    This does not create a public bucket if one does not exist.
+    This fails if the bucket does not exist or if it's private.
+    """
+    msgs = []
+    auth = identity.v3.Password(**self.creds.provide_keystone_v3())
+    session_ = session.Session(auth=auth)
+    keystone = keystoneclientv3.Client(session=session_)
+
+    project_id = keystone.projects.client.get_project_id()
+
+    if project_id == None:
+      raise ProjectNotAvailableException(msgs=self.tenant)
+
+    swift_url = "{}/AUTH_{}/{}".format(self.swift_publicurl,project_id,self.swift_bucket)
+
+    _html = requests.get(swift_url).content
```

Comment:
I gave a bad review for this yesterday. Everything works perfectly with urllib as long as one uses `nrpe` user's environment.

We have a fair amount of historical data from the S3 public check, and all that has been done with urllib. Using another library with Swift public check will make the data less comparable across S3<->Swift. Please revert back to urllib.

Reply:
No worries. I have reverted the changes.